internal/git/gitsync_processor.go

```diff
@@ -139,10 +162,23 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1.GitSync, k8client clie
 		gitCh := make(chan Message, messageChanLength)
 		channels[repo.Name] = gitCh
 		go func(repo *v1.RepositoryPath) {
-			err := watchRepo(ctx, repo, namespace)
+			r, err := cloneRepo(repo)
 			if err != nil {
-				// TODO: Retry on non-fatal errors
-				logger.Errorw("error watching the repo", "err", err)
+				logger.Errorw("error cloning the repo",
+					"GitSync name", gitSync.Name,
+					"RepositoryPath Name", repo.Name,
+					"err", err)
+			} else {
+				for ok := true; ok; {
+					err = watchRepo(ctx, r, gitSync, k8client, repo, namespace)
+					// TODO: Break on fatal errors
```

Comment:
If there were a fatal error in this goroutine (which I'm not sure if there would be?), wouldn't the inner code just directly call `log.Fatal()` and then the whole thing would shut down? Are you thinking we need to check for it here for some reason?

Reply:
Sorry I should make it more clear the fatal errors I put here is for the GitSyncProcessor goroutine, not the entire process should shut down in this case.